server/version.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+// VersionFeature is a unique identifier for minimum support version.
+type VersionFeature int
+
+// Version Fetures.
+const (
+	VersionBase VersionFeature = iota
+	VersionRegionMergeAndRaftLearner
```

Comment:
em, too long...

Btw, they are two different things, why not using two version?

Reply:
Yes, I would suggest changing from `VersionFeature` to just `Feature` and here splitting these into two features. The featuresDict can map them both to the same version. I would also remove `VersionBase`.